golos.charge/golos.charge.cpp

```diff
@@ -50,15 +49,23 @@ fixp_t charge::consume_charge(name issuer, name user, symbol_code token_code, ui
 }
 
 void charge::use(name user, symbol_code token_code, uint8_t charge_id, int64_t price_arg, int64_t cutoff_arg, int64_t vesting_price) {
-    consume_charge(token::get_issuer(config::token_name, token_code), user, token_code, charge_id, price_arg, cutoff_arg, vesting_price);
+    auto issuer = token::get_issuer(config::token_name, token_code);
+    require_auth(issuer);
+    consume_charge(issuer, user, token_code, charge_id, price_arg, cutoff_arg, vesting_price);
+}
+
+void charge::check(name user, symbol_code token_code, uint8_t charge_id, int64_t cutoff_arg) {
+    //require_auth(anyone); this action can update the value, but only by a predetermined restorer
+    consume_charge(token::get_issuer(config::token_name, token_code), user, token_code, charge_id, 0, cutoff_arg, 0);
```

Comment:
it's better to have read-only function for this. is it possible to extract additional read-only function like `get_current_charge` from `consume_charge` and use it inside `consume_charge` and here?

Reply:
there is a restorer here, it is performed only if some time has passed since the last calculation. therefore we keep a value after restoring so that it can be used in the same block. do you think it is better to recalculate the value in such cases, but provide a read-only check?

Reply:
you're right, read-only is better anyway